### PR TITLE
Introduce a validation set for S3DIS

### PR DESCRIPTION
**Pull request by nicolas-chaulet**
Training command line:
```
python train.py task=segmentation model_type=pointnet2 model_name=pointnet2_largemsg dataset=s3disfused-fixed training=s3dis_benchmark/pointnet2

python train.py task=segmentation model_type=kpconv model_name=KPConvPaper dataset=s3disfused training=s3dis_benchmark/kpconv

python train.py task=segmentation model_type=rsconv model_name=RSConv_MSN_S3DIS dataset=s3disfused-fixed training=s3dis_benchmark/rsconv
```

### CODE REVIEW

File: torch_points3d/datasets/segmentation/s3dis.py [hunk 1]
```diff
@@ -384,28 +415,50 @@ def process(self):
...
+            if i != self.test_area - 1:
+                train_data_list[i] = []
+                val_data_list[i] = []
+                trainval_data_list[i] = []
+                for data in data_list[i]:
+                    validation_set = data.validation_set
+                    del data.validation_set
+                    if validation_set:
+                        val_data_list[i].append(data)
+                        trainval_data_list[i].append(data)
```
Comment (tchaton) [inline torch_points3d/datasets/segmentation/s3dis.py:438]: trainval_data_list could have been the concatenation of train + val

trainval_data_list = train_data_list + val_data_list

Should simplify the code.

File: torch_points3d/datasets/segmentation/s3dis.py [hunk 2]
```diff
@@ -457,6 +510,12 @@ def get(self, idx):
         else:
             return self._test_spheres[idx]
 
+    def process(self):
```
Comment (tchaton) [inline torch_points3d/datasets/segmentation/s3dis.py:513]: Why did you add those ?

File: torch_points3d/datasets/segmentation/s3dis.py (hunk 1 above)
Comment (nicolas-chaulet) [reply torch_points3d/datasets/segmentation/s3dis.py:438]: Good point, changing that

File: torch_points3d/datasets/segmentation/s3dis.py (hunk 2 above)
Comment (nicolas-chaulet) [reply torch_points3d/datasets/segmentation/s3dis.py:513]: Pytorch geometric tests if the function exists. If it does not then it does not run those methods... So I had to add them. I'll add a comment

**Comment by guochengqian** [comment]
Great work! Could your code reproduce the performance of KPConv paper?
